tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp

```diff
@@ -1347,16 +1377,29 @@ SQLITE_API void sqlite3_result_error_nomem(sqlite3_context *) {
 SQLITE_API void sqlite3_result_error_code(sqlite3_context *, int) {
 }
 
-SQLITE_API void sqlite3_result_int(sqlite3_context *, int) {
+SQLITE_API void sqlite3_result_int(sqlite3_context *context, int val) {
+	context->result.u.i = val;
+	context->result.type = SQLiteTypeValue::INTEGER;
 }
 
 SQLITE_API void sqlite3_result_int64(sqlite3_context *, sqlite3_int64) {
 }
 
-SQLITE_API void sqlite3_result_null(sqlite3_context *) {
+SQLITE_API void sqlite3_result_null(sqlite3_context *context) {
+	context->result.type = SQLiteTypeValue::NULL_VALUE;
 }
 
-SQLITE_API void sqlite3_result_text(sqlite3_context *, const char *, int, void (*)(void *)) {
+SQLITE_API void sqlite3_result_text(sqlite3_context *context, const char *str_c, int n_chars, void (*)(void *)) {
+	if (str_c) {
+		auto utf_type = Utf8Proc::Analyze(str_c, n_chars);
+		if (utf_type == UnicodeType::INVALID) {
+			throw Exception("Text value is not valid UTF8");
```

Comment:
You should not throw exceptions from within a C API function. C programs cannot deal with exceptions as they are a C++ function, so this will always result in unpredictable behavior and/or aborting the program.